js/controller.js

```diff
@@ -66,6 +66,32 @@
             SoundCloudService.init();
 
             var refreshMirrorData = function() {
+                CalendarService.getCalendarEvents().then(function(response) {
+                    $scope.calendar = CalendarService.getFutureEvents();
+                }, function(error) {
+                    console.log(error);
+                });
+
+                if ($scope.fitbitEnabled) {
+                    setTimeout(function() { refreshFitbitData(); }, 5000);
```

Comment:
Yeah. Really each of the mirror components should be directives and self modify their scope and state.